tool/tsh/tsh.go

```diff
@@ -783,6 +823,22 @@ func onSSH(cf *CLIConf) {
 		return tc.SSH(cf.Context, cf.RemoteCommand, cf.LocalExec)
 	})
 	if err != nil {
+		if strings.Contains(utils.UserMessageFromError(err), teleport.ServerIsAmbiguous) {
+			allNodes, err := tc.ListAllNodes(cf.Context)
+			if err != nil {
+				utils.FatalError(err)
+			}
+			var nodes []services.Server
+			for _, node := range allNodes {
+				if node.GetHostname() == tc.Host {
+					nodes = append(nodes, node)
+				}
+			}
+			fmt.Fprintf(os.Stderr, "error: ambiguous host could match multiple nodes\n\n")
+			showNodes(nodes, true)
+			fmt.Fprintf(os.Stderr, "Hint: try addressing the node by unique id (ex: tctl ssh user@node-id)\n")
```

Comment:
Maybe we can add a hint here to tell people how to do this again if they need to, something like: "To show node UUIDs in future, run `'tsh ls -v'`" (or whatever the command is)

(and if we aren't exposing this via a flag to `tsh ls`, I think we should be)